### ISSUE TITLE
Activate virtualenv if Ansible is installed in one

### ISSUE THREAD

**Issue by mdklatt** [mdklatt/idea-ansible-plugin#2]
It is common to run Ansible inside a Python [virtualenv](https://virtualenv.pypa.io/en/latest) environment. Typically, the envrionment is activated before running any commands inside it. It is possible to explicitly call `python` or another executable from the environment's bin directory without activation, *e.g.* `venv/bin/ansible-playbook`, and this will work most cases because the main thing activation does is set `$PATH` and `$PYTHONPATH`. However, this is not guaranteed to work, so best practice is to either activate the environment first or in a one-liner: `source venv/bin/activate && ansible-playbook ...` 

There are known cases where failing to activate the environment can cause complications in an Ansible playbook. Assume the following play is executed using Ansible in a virtualenv. In the run configuration, the Ansible command is set to the explicit virtualenv path: `venv/bin/ansible-playbook`, which normally works. However, when the play attempts to run `ansible-galaxy` from the same virtualenv, it cannot because `$PATH` is not set correctly. A workaround would be to explicitly set the path to match the parent `ansible-playbook` command, but this is inconvenient at best.

``` yaml
---
- name: Resolve Ansible dependencies
  hosts: localhost
  connection: local
  gather_facts: false

  tasks:
    - name: Install requirements
      run_once: true
      block:
        - name: Install Ansible roles
          command: "ansible-galaxy role install -r requirements.yml -p roles"
        - name: Install Ansible collections
          command: "ansible-galaxy collection install -r requirements.yml -p collections"
```

The solution is to add the ability to both the Playbook and Galaxy run configurations to indicate that the command is running inside a virtualenv environment that needs activation. This would also be useful for [idea-remotepython-plugin](https://github.com/mdklatt/idea-remotepython-plugin), which makes the core functionality a candidate for inclusion in [idea-common](https://github.com/mdklatt/idea-common)--something like a `runVenvCommand()` function.

**Comment by mdklatt** [comment]
Per the [virtualenv documenation](https://virtualenv.pypa.io/en/latest/user_guide.html#activators), activation is merely a convenience for setting `$PATH`. Upon closer inspection of the activation script, it does *not* set `$PYTHONPATH` as claimed above. It does, however, set `$VIRTUAL_ENV` to the environment root directory, and makes sure `$PYTHONHOME` is not set. Deactivation restores `$PATH`, `$PYTHONHOME` if it was set, and unsets `$VIRTUAL_ENV`.

Modifying `$PATH` and unsetting `$PYTHONPATH` in the process's child environment is trivial, and deactivation is not required in this case because the parent environment is never modified. Is setting `$VIRTUAL_ENV` necessary? It might be used by Ansible or other applications to detect if they are running in a virtualenv.

**Comment by mdklatt** [comment]
Commit 4e70944e548ee6200224084d32b62e90998c38ee is available for testing with real-world projects.

**Comment by mdklatt** [comment]
Issue resolved by commit 4e70944e548ee6200224084d32b62e90998c38ee.
